### PR TITLE
Homework C++ fundamentals password-check

### PR DESCRIPTION
**Pull request by gyrydymptum**
Authors: @mariuszlisowski, @gyrydymptum

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @mariuszlisowski

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @gyrydymptum

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @mariuszlisowski

**Comment by ziobron** [comment]
⏱ 3 XP extra for @mariuszlisowski for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @gyrydymptum

**Comment by ziobron** [comment]
⏱ 3 XP extra for @gyrydymptum for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @mariuszlisowski

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @gyrydymptum

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+    }
+    if (!(std::any_of(password.begin(), password.end(), ::isdigit))) {
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    }
+    if (!(std::any_of(password.begin(), password.end(), ::ispunct))) {
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    }
+    if (!(std::any_of(password.begin(), password.end(), ::isupper))) {
+        return ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter;
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:40]: better use none_of

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**